orchagent/portsorch.cpp

```diff
@@ -243,6 +243,77 @@ bool PortsOrch::setPortMtu(sai_object_id_t id, sai_uint32_t mtu)
     return true;
 }
 
+bool PortsOrch::validatePortSpeed(sai_object_id_t port_id, sai_uint32_t speed)
+{
+    sai_attribute_t attr;
+    sai_status_t status;
+
+    // validate speed value
+    attr.id = SAI_PORT_ATTR_SUPPORTED_SPEED;
+    attr.value.u32list.count = 0;
+    attr.value.u32list.list = NULL;
+    status = sai_port_api->get_port_attribute(port_id, 1, &attr);
+    if (status == SAI_STATUS_BUFFER_OVERFLOW)
+    {
+        uint32_t speeds[attr.value.u32list.count];
```

Comment:
no compilation error (C99) but not recommended by the community. Will rewrite to std::vector